services/viva-ms/lambdas/createVivaCase.js

```diff
@@ -169,14 +177,64 @@ async function putRecurringVivaCase(PK, workflowId, period) {
     },
   };
 
-  const [putItemError, caseItem] = await to(putItem(putItemParams));
+  const casePersonList = getCasePersonList(vivaPerson);
+  caseItemPutParams.Item['persons'] = casePersonList;
+
+  const casePersonCoApplicant = getUserByRole(casePersonList, 'coApplicant');
+  if (casePersonCoApplicant) {
+    caseItemPutParams.Item['GSI1'] = `USER#${casePersonCoApplicant.personalNumber}`;
+  }
+
+  const [putItemError, caseItem] = await to(putItem(caseItemPutParams));
   if (putItemError) {
     throw putItemError;
   }
 
   return caseItem;
 }
 
+function getUserByRole(userList, role) {
+  const user = userList.find(user => user.role == role);
+  return user;
+}
+
+function getCasePersonList(vivaPerson) {
+  const person = vivaPerson.case.persons?.person;
+  const client = vivaPerson.case.client;
+  client['type'] = 'client';
+
+  const vivaPersonList = [];
+  vivaPersonList.push(client);
+
+  if (Array.isArray(person)) {
+    vivaPersonList.push(...person);
+  } else if (person != undefined) {
+    vivaPersonList.push(person);
+  }
+
+  const roleTranslateList = {
+    client: 'applicant',
+    partner: 'coApplicant',
+    child: 'children',
+  };
+
+  const casePersonList = vivaPersonList.map(person => {
+    const { pnumber, fname: firstName, lname: lastName, type } = person;
+    const personalNumber = String(pnumber).replace(/\D/g, '');
+
+    const role = Object.keys(roleTranslateList).includes(type) && roleTranslateList[type];
```

Comment:
If type is invalid then role becomes a bool (`false`), and `<bool> && <string|undefined>` is not very clear, given javascript's equality inconsistency.

Suggestion: change to `const role = roleTranslateList[type] || ""`.